cli/cli-client/src/main/java/org/infinispan/cli/impl/ContextImpl.java

```diff
@@ -59,14 +82,41 @@ public boolean isConnected() {
 
    @Override
    public void setProperty(String key, String value) {
-      properties.setProperty(key, value);
+      if (value == null) {
+         properties.remove(key);
+      } else {
+         properties.setProperty(key, value);
+      }
    }
 
    @Override
    public String getProperty(String key) {
       return properties.getProperty(key);
    }
 
+   @Override
+   public String getProperty(Property property) {
+      return properties.getProperty(property.propertyName());
+   }
+
+   @Override
+   public Properties getProperties() {
+      return properties;
+   }
+
+   @Override
+   public void saveProperties() {
+      Path configFile = configPath.resolve(CONFIG_FILE);
+      try {
+         Files.createDirectories(configPath);
+         Writer w = Files.newBufferedWriter(configFile);
+         properties.store(w, null);
+         Util.close(w);
```

Comment:
This should either be in a finally block or we should use a try-with-resources when creating the Writer.

Reply:
Done.